orchestrate/src/vm/error.rs

```diff
@@ -38,7 +38,16 @@ impl wasmi::core::HostError for VmError {}
 
 impl From<wasmi::Error> for VmError {
     fn from(e: wasmi::Error) -> Self {
-        Self::Interpreter(format!("{e}"))
+        match e {
+            wasmi::Error::Trap(ref trap) => {
+                if let Some(err) = trap.downcast_ref::<VmError>() {
```

Comment:
Here you assume it's a VmError while we allow the user to provide an error that CONTAINS VmError, I'm not sure it will be working, did you give it a try?

Reply:
Host functions are returning `VmErrorOf<V>` which is the error type of the underlying vm. In this case it is `VmError`. Isn't it correct? I also returned early errors from vm functions and it worked.

Reply:
If you have a VM, it implements an error type `E` which has a `From<WasmiVMError> + From<BlablaErr...>` which means you would catch a trapped `E` here, how do you know what `E` is in this context? You can't know the error so it's not VmError.